user_modules/shared_testcases/testCasesForPolicyTable.lua

```diff
@@ -824,5 +834,230 @@ function testCasesForPolicyTable:Restore_preloaded_pt()
+	      :Do(function(_,data)
+			
+		    EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate", {status = "UPDATING"})
+		    :Do(function(_,data) 
+		    end)--EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate", {status = "UPDATING"})
+
+			local CorIdSystemRequest = self.mobileSession:SendRPC("SystemRequest",  {requestType = "PROPRIETARY", fileName = ptu_file_name}, ptu_file)
+			EXPECT_HMICALL("BasicCommunication.SystemRequest",{ requestType = "PROPRIETARY", fileName = SystemFilesPath..ptu_file_name })
+            :Do(function(_,data) 
+            	self.hmiConnection:SendResponse(systemRequestId,"BasicCommunication.SystemRequest", "SUCCESS", {})
+            	self.hmiConnection:SendNotification("SDL.OnReceivedPolicyUpdate", { policyfile = SystemFilesPath..ptu_file_name})
+            	EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate", {status = "UP_TO_DATE"})
+            end) --EXPECT_HMICALL("BasicCommunication.SystemRequest",{ requestType = "PROPRIETARY", fileName = ptu_file })
+            --TODO(istoimenova): SDL returns GENERIC_ERROR. Should be checked. As not in scope will be commented
+			--EXPECT_RESPONSE(CorIdSystemRequest, { success = true, resultCode = "SUCCESS"})
+		  end) --EXPECT_NOTIFICATION("OnSystemRequest", {requestType = "PROPRIETARY"})
+        end) --EXPECT_HMIRESPONSE(RequestId_GetUrls,{result = {code = 0, method = "SDL.GetURLS", urls = endpoints} } )
+    end -- function Test:Flow_PTU_SUCCEESS_EXTERNAL_PROPRIETARY()
+end
+
+function testCasesForPolicyTable:trigger_PTU_user_request_update_from_HMI()
+	--function is created only for one app due to luck of time should not be updated at the moment.
+	function Test:TestStep_trigger_PTU_user_request_update_from_HMI()
+		local hmi_app1_id = self.applications[config.application1.registerAppInterfaceParams.appName]
+		testCasesForPolicyTable.time_trigger = 0
+		testCasesForPolicyTable.time_onstatusupdate = 0
+		testCasesForPolicyTable.time_policyupdate = 0
+
+		self.hmiConnection:SendNotification("SDL.OnPolicyUpdate", {} )
+		
+		testCasesForPolicyTable.time_trigger = timestamp()
+
+	    EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate", {status = "UPDATE_NEEDED"})
+	    :Do(function(_,data) 
+	    	testCasesForPolicyTable.time_onstatusupdate = timestamp()
+	    end)
+
+	    testCasesForPolicyTableSnapshot:create_PTS(true, {
+          config.application1.registerAppInterfaceParams.appID
+        },
+        {config.deviceMAC},
+        {hmi_app1_id})
+	      
+	    local timeout_after_x_seconds = testCasesForPolicyTableSnapshot:get_data_from_PTS("module_config.timeout_after_x_seconds")
+	    local seconds_between_retries = {}
+	    for i = 1, #testCasesForPolicyTableSnapshot.pts_seconds_between_retries do
+	      seconds_between_retries[i] = testCasesForPolicyTableSnapshot.pts_seconds_between_retries[i].value
+	    end
+	    EXPECT_HMICALL("BasicCommunication.PolicyUpdate", 
+	        {
+	          file = "/tmp/fs/mp/images/ivsu_cache/sdl_snapshot.json", 
```

Comment:
@Itileda I didn't manage to find function that only reads data from INI file. All functions that are implemented till now are updating INI file.
It is almost added in google sheet that such function should be prepared(commonSteps:get_data_form_SDL_ini), but due to luck of time and not so major for now I still didn't created. In this way  I use hardcoded values from INI file.